core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -119,4 +124,22 @@ private Mappers() {
             throw new RuntimeException( e );
         }
     }
+
+    /**
+     * <p>
+     * Returns the simple name of a class with all enclosing classes' simple names prepended,
+     * separated by an underscore.
+     * </p>
+     * <p>
+     * Example: returns <code>Map_Entry</code> for {@link java.util.Map.Entry}.
+     * </p>
+     */
+    public static String getFlatName(Class<?> klass) {
+        String flatName = klass.getSimpleName();
+        while (klass.isMemberClass()) {
+            klass = klass.getEnclosingClass();
+            flatName = klass.getSimpleName() + "_" + flatName;
```

Comment:
+1 for using a string builder in case the class is an inner one (i.e. let's avoid the string builder for the default case of a top-level class).